internal/manifests/queryfrontend/query_frontend.go

```diff
@@ -64,12 +64,14 @@ func BuildQueryFrontend(params manifestutils.Params) ([]client.Object, error) {
 		manifests = append(manifests, s)
 	}
 
-	//exhaustive:ignore
-	switch tempo.Spec.Template.QueryFrontend.JaegerQuery.Ingress.Type {
-	case v1alpha1.IngressTypeIngress:
-		manifests = append(manifests, ingress(tempo))
-	case v1alpha1.IngressTypeRoute:
-		manifests = append(manifests, route(tempo))
+	if !tempo.Spec.Template.Gateway.Enabled {
```

Comment:
In case the user mistakenly enables both, now the ingress/route silently doesn't get created and the user might be left wondering why. Could you add a validation in the webhook to explicitly tell the user that he/she cannot enable ingress/route and gateway at the same time?

Reply:
Good point. I'll add a validation.

Reply:
Added